javaparser-core/src/main/java/com/github/javaparser/ast/body/VariableDeclarator.java

```diff
@@ -151,8 +157,24 @@ public Type getType() {
     public VariableDeclarator setType(final Type type) {
         assertNotNull(type);
         notifyPropertyChange(ObservableProperty.TYPE, this.type, type);
-        if (this.type != null)
+        if (this.getParentNode().isPresent() && this.getParentNode().get() instanceof NodeWithVariables) {
+            NodeWithVariables nodeWithVariables = (NodeWithVariables)this.getParentNode().get();
+            Type currentMaxCommonType = nodeWithVariables.getMaximumCommonType();
+            List<Type> types = new LinkedList<>();
```

Comment:
Noted: I will find a workaround and extract this as a separate change so we can look into it without all the other stuff